lib/feedx/pusher.rb

```diff
@@ -62,11 +69,10 @@ def detect_compress(val)
       end
     end
 
-    def write_to(io)
+    def write_enum(enum, io)
       stream   = @format.new(io)
-      relation = @enum.is_a?(Proc) ? @enum.call : @enum
-      iterator = relation.respond_to?(:find_each) ? :find_each : :each
-      relation.send(iterator) {|rec| stream.write(rec) }
+      iterator = enum.respond_to?(:find_each) ? :find_each : :each
+      enum.send(iterator) {|rec| stream.write(rec) }
```

Comment:
There's no caching for `def enum`, it'll be resolved twice here.
I'd suggest to somehow cache `self.enum` value to be more consistent.

Reply:
it's cached, it's not using `self.enum` but the passed argument, I can make it nicer by renaming it though